csharp/src/Ice/Current.cs

```diff
@@ -8,36 +8,27 @@ namespace Ice
 
     public class Current
     {
-        public ObjectAdapter adapter;
-        public Connection con;
-        public Identity id;
-        public string facet;
-        public string operation;
-        public OperationMode mode;
-        public Dictionary<string, string> ctx;
-        public int requestId;
-        public EncodingVersion encoding;
+        public ObjectAdapter Adapter { get; private set; }
+        public Connection? Connection { get; private set; }
+        public Identity Id { get; private set; }
+        public string Facet { get; private set; }
+        public string Operation { get; private set; }
+        public OperationMode Mode { get; private set; }
+        public Dictionary<string, string>? Context { get; private set; }
+        public int RequestId { get; private set; }
+        public EncodingVersion Encoding { get; internal set; }
 
-        public Current()
+        public Current(ObjectAdapter adapter, Identity id, string facet, string operation,
+            OperationMode mode, Dictionary<string, string> ctx, int requestId, Connection? conn)
```

Comment:
The Context property and ctx ctor parameter should have the same type, probably non-nullable Dictionary<string, string>

Reply:
Do we need to differentiate between non set context and empty context?

Reply:
There is no distinction. With the ice1 protocol, the context is always sent and can be empty.